### ISSUE TITLE
[bug] Z2M not reacting to notification from device after re-add

### ISSUE THREAD

**Issue by LordMike** [zwave-js/zwavejs2mqtt#389]
### Version

Build/Run method

- [ ] Docker
- [ ] PKG
- [x] Manually built (git clone - npm install - npm run build )

zwavejs2mqtt version:  1.0.0-rc.1+441fc7e.441fc7e 
zwavejs version:  6.1.0-f0eff3e 

### Describe the bug
Strangest thing. I have two ZWA008 Door Windows Sensor 7's on my desk, both sending the same frames when their sensor is triggered (magnet). But for one of them, Zwavejs2mqtt does _not_ react to it.

In the logs below, Node 060 is the working device, and is swiped first. Zwavejs2mqtt gets the two notifications without any issue - but immediately after, when swiping Node 024, nothing happens in zwavejs2mqtt..

Update: I realized that earlier this evening I re-added node 24 a few times, as I didn't see it get added (I expected node 61). So I now know how to reproduce it.

### To Reproduce
~~I'm not sure.~~

1. Have included device (node 24 in my example)
1. Start inclusion
2. Triple-click the ZWA008 (node 24) to start its inclusion process, zwavejs2mqtt will log the error (see below)
3. Future notifications from ZWA008 does not arrive in mqtt

### Expected behavior
Re-adding existing nodes does not prevent future notifications from working.

### Additional context

This is the log as I add the device the second time.

```
zwavejs2mqtt_1  | 2021-01-29 20:50:41.809 INFO ZWAVE: Node 24: added
zwavejs2mqtt_1  | 2021-01-29 20:50:41.811 ERROR ZWAVE: Error while adding node 24 node has been added twice
zwavejs2mqtt_1  | Error: node has been added twice
zwavejs2mqtt_1  |     at ZwaveClient.addNode (/usr/src/app/lib/ZwaveClient.js:749:7)
zwavejs2mqtt_1  |     at ZwaveClient.onNodeAdded (/usr/src/app/lib/ZwaveClient.js:242:13)
zwavejs2mqtt_1  |     at ZWaveController.emit (events.js:326:22)
zwavejs2mqtt_1  |     at ZWaveController.EventEmitter.emit (domain.js:483:12)
zwavejs2mqtt_1  |     at ZWaveController.handleAddNodeRequest (/usr/src/app/node_modules/zwave-js/src/lib/controller/Controller.ts:998:11)
zwavejs2mqtt_1  |     at Driver.handleRequest (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1902:33)
zwavejs2mqtt_1  |     at Driver.handleUnsolicitedMessage (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1638:16)
zwavejs2mqtt_1  |     at Object.notifyUnsolicited (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:487:16)
zwavejs2mqtt_1  |     at notifyUnsolicited (/usr/src/app/node_modules/zwave-js/src/lib/driver/SendThreadMachine.ts:563:19)
zwavejs2mqtt_1  |     at Interpreter.exec (/usr/src/app/node_modules/xstate/lib/interpreter.js:653:24)
zwavejs2mqtt_1  |     at Interpreter.execute (/usr/src/app/node_modules/xstate/lib/interpreter.js:206:22)
zwavejs2mqtt_1  |     at Interpreter.update (/usr/src/app/node_modules/xstate/lib/interpreter.js:226:18)
zwavejs2mqtt_1  |     at /usr/src/app/node_modules/xstate/lib/interpreter.js:125:23
zwavejs2mqtt_1  |     at Scheduler.process (/usr/src/app/node_modules/xstate/lib/scheduler.js:60:13)
zwavejs2mqtt_1  |     at Scheduler.flushEvents (/usr/src/app/node_modules/xstate/lib/scheduler.js:53:18)
zwavejs2mqtt_1  |     at Scheduler.schedule (/usr/src/app/node_modules/xstate/lib/scheduler.js:45:14)
zwavejs2mqtt_1  | 2021-01-29 20:50:41.815 INFO ZWAVE: Controller status: Inclusion stopped
```

#### Original logs, showing that zwavejs2mqtt does not react to events

These are the logs from the two projects, as the magnet is swiped across both in sequence.

zwavejs2mqtt
```
zwavejs2mqtt_1  | 2021-01-29 20:45:03.346 INFO ZWAVE: Node 4 is now asleep
zwavejs2mqtt_1  | 2021-01-29 20:45:28.748 INFO ZWAVE: Node 60: value updated: 113-0-Access Control-Door state 22 => 23
zwavejs2mqtt_1  | 2021-01-29 20:45:29.013 INFO ZWAVE: Node 60: value updated: 113-0-Access Control-Door state 23 => 22
```

zwave-js (spacing is mine)
```
20:45:03.327 SERIAL « 0x0104011301e8                                                       (6 bytes)
20:45:03.327 SERIAL » [ACK]                                                                   (0x06)
20:45:03.327 DRIVER « [RES] [SendData]
                        was sent: true
20:45:03.343 SERIAL « 0x01070013ff00000216                                                 (9 bytes)
20:45:03.343 SERIAL » [ACK]                                                                   (0x06)
20:45:03.343 DRIVER « [REQ] [SendData]
                        callback id:     255
                        transmit status: OK
20:45:03.347 CNTRLR   [Node 004] The node is now asleep.


20:45:28.744 SERIAL « 0x010f0004003c097105000000ff0617005b                                (17 bytes)
20:45:28.744 SERIAL » [ACK]                                                                   (0x06)
20:45:28.745 DRIVER « [Node 060] [REQ] [ApplicationCommand]
                      └─[NotificationCCReport]
                          notification type:   Access Control
                          notification status: 255
                          notification state:  Window/door is closed
20:45:28.747 CNTRLR   [Node 060] [~] [Notification] Access Control[Door state]: 22 => 2 [Endpoint 0]
                      3
20:45:29.011 SERIAL « 0x010f0004003c097105000000ff0616005a                                (17 bytes)
20:45:29.011 SERIAL » [ACK]                                                                   (0x06)
20:45:29.012 DRIVER « [Node 060] [REQ] [ApplicationCommand]
                      └─[NotificationCCReport]
                          notification type:   Access Control
                          notification status: 255
                          notification state:  Window/door is open
20:45:29.013 CNTRLR   [Node 060] [~] [Notification] Access Control[Door state]: 23 => 2 [Endpoint 0]
                      2



20:45:31.963 SERIAL « 0x010f00040018097105000000ff0617007f                                (17 bytes)
20:45:31.964 SERIAL » [ACK]                                                                   (0x06)
20:45:31.964 DRIVER « [Node 024] [REQ] [ApplicationCommand]
                      └─[NotificationCCReport]
                          notification type:   Access Control
                          notification status: 255
                          notification state:  Window/door is closed
20:45:31.965 CNTRLR   [Node 024] [~] [Notification] Access Control[Door state]: 22 => 2 [Endpoint 0]
                      3
20:45:31.981 SERIAL « 0x01090004001803200100c8                                            (11 bytes)
20:45:31.982 SERIAL » [ACK]                                                                   (0x06)
20:45:31.982 DRIVER « [Node 024] [REQ] [ApplicationCommand]
                      └─[BasicCCSet]
                          target value: 0
20:45:31.983 CNTRLR   [Node 024] treating BasicCC Set as a report
20:45:31.983 CNTRLR   [Node 024] [~] [Basic] currentValue: 255 => 0                     [Endpoint 0]
20:45:32.031 SERIAL « 0x010f00040018097105000000ff0616007e                                (17 bytes)
20:45:32.031 SERIAL » [ACK]                                                                   (0x06)
20:45:32.032 DRIVER « [Node 024] [REQ] [ApplicationCommand]
                      └─[NotificationCCReport]
                          notification type:   Access Control
                          notification status: 255
                          notification state:  Window/door is open
20:45:32.032 CNTRLR   [Node 024] [~] [Notification] Access Control[Door state]: 23 => 2 [Endpoint 0]
                      2
20:45:32.049 SERIAL « 0x010900040018032001ff37                                            (11 bytes)
20:45:32.050 SERIAL » [ACK]                                                                   (0x06)
20:45:32.050 DRIVER « [Node 024] [REQ] [ApplicationCommand]
                      └─[BasicCCSet]
                          target value: 255
20:45:32.051 CNTRLR   [Node 024] treating BasicCC Set as a report
20:45:32.052 CNTRLR   [Node 024] [~] [Basic] currentValue: 0 => 255                     [Endpoint 0]

```

**Comment by LordMike** [comment]
Rebooting the server resolves the immediate issue - zwavejs2mqtt is now receiving events.

**Comment by robertsLando** [comment]
@LordMike When you say re-adding what you exaclty mean? Did you also excluded the device? Do you mean you have re-interview it?

**Comment by LordMike** [comment]
None of the above. The steps would be:

1. Include the device (from scratch - or pick any existing device)
2. Put z-wave into include mode
3. Do steps on the device to enable inclusion/exclusion.
4. Observe error

For my test, I used the Aeotec door window sensor 7. It has a switch on the side, that if you tap it three times, it can be included or excluded.

**Comment by robertsLando** [comment]
Still don't get it

**Comment by LordMike** [comment]
What if you try with one of your devices?

* Where you'd normally put the network in exclude mode, and do something with the device (tap it, wake it, ... whatever is needed)
* Put the network in _include mode instead_

I think maybe the network frames for include/exclude is the same, so it completely depends on the controller, what the resulting action is. If the device was able to communicate that it wanted to be included, the controller could see that it was in exclude mode, and thus not try to include the device - or vice versa.

@AlCalzone is it possible to know if a device being included, has been included before, or maybe .. should this bug report actually be in zwave-js?

**Comment by AlCalzone** [comment]
> I think maybe the network frames for include/exclude is the same, so it completely depends on the controller, what the resulting action is.

That! We only get a notification that a node was added along with its node ID. I didn't think that including an already included device would work, but it seems that it does. `node-zwave-js` just overwrites the old node instance. A few timers might not get cleaned up in this case, but I don't think this is too critical, since it is still the same device.

Since the error comes from zwavejs2mqtt, this issue should be handled there. I think the node should just be handled like any other new one.

**Comment by LordMike** [comment]
The add process fails with the exception shown above - `Error while adding node 24 node has been added twice`. I suspect zwavejs2mqtt doesn't register some handler, when that is raised... But should that error come in this case - if it's to be handled as you say?

**Comment by robertsLando** [comment]
@LordMike That error is thrown on my side because if the node is added twice it's an error for me

**Comment by AlCalzone** [comment]
@LordMike if you look at the stack trace, you'll see that the location of the error is **not** located in `node_modules/zwave-js` or `node_modules/@zwave-js`:
```
--- zwavejs2mqtt ---
zwavejs2mqtt_1  | Error: node has been added twice
zwavejs2mqtt_1  |     at ZwaveClient.addNode (/usr/src/app/lib/ZwaveClient.js:749:7)
zwavejs2mqtt_1  |     at ZwaveClient.onNodeAdded (/usr/src/app/lib/ZwaveClient.js:242:13)

--- Node.js internals (EventEmitter) ---
zwavejs2mqtt_1  |     at ZWaveController.emit (events.js:326:22)
zwavejs2mqtt_1  |     at ZWaveController.EventEmitter.emit (domain.js:483:12)

--- node-zwave-js ---
zwavejs2mqtt_1  |     at ZWaveController.handleAddNodeRequest (/usr/src/app/node_modules/zwave-js/src/lib/controller/Controller.ts:998:11)
zwavejs2mqtt_1  |     at Driver.handleRequest (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1902:33)
zwavejs2mqtt_1  |     at Driver.handleUnsolicitedMessage (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1638:16)
zwavejs2mqtt_1  |     at Object.notifyUnsolicited (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:487:16)
zwavejs2mqtt_1  |     at notifyUnsolicited (/usr/src/app/node_modules/zwave-js/src/lib/driver/SendThreadMachine.ts:563:19)

--- xstate (zwave-js's state machine) ---
zwavejs2mqtt_1  |     at Interpreter.exec (/usr/src/app/node_modules/xstate/lib/interpreter.js:653:24)
zwavejs2mqtt_1  |     at Interpreter.execute (/usr/src/app/node_modules/xstate/lib/interpreter.js:206:22)
zwavejs2mqtt_1  |     at Interpreter.update (/usr/src/app/node_modules/xstate/lib/interpreter.js:226:18)
zwavejs2mqtt_1  |     at /usr/src/app/node_modules/xstate/lib/interpreter.js:125:23
zwavejs2mqtt_1  |     at Scheduler.process (/usr/src/app/node_modules/xstate/lib/scheduler.js:60:13)
zwavejs2mqtt_1  |     at Scheduler.flushEvents (/usr/src/app/node_modules/xstate/lib/scheduler.js:53:18)
zwavejs2mqtt_1  |     at Scheduler.schedule (/usr/src/app/node_modules/xstate/lib/scheduler.js:45:14)
```

**Comment by robertsLando** [comment]
@LordMike is this still an issue?

**Comment by LordMike** [comment]
> @LordMike is this still an issue?

I'll check later today.

**Comment by LordMike** [comment]
Just discovered that `:dev` is no longer supported.. So I've just been hit by a bunch of new stuff :)

Versions (fixed a typo in #1135):
```
zwavejs2mqtt: 4.0.1
zwave-js: 7.3.0
home id: 3815907651
home hex: 0xe3722143
```

I can confirm that the issue is still present. The debug log shows:

```
2021-05-04 17:04:34.552 INFO ZWAVE: Node 36: added
2021-05-04 17:04:34.554 ERROR ZWAVE: Error while adding node 36 node has been added twice
Error: node has been added twice
at ZwaveClient.addNode (/usr/src/app/lib/ZwaveClient.js:873:7)
at ZwaveClient.onNodeAdded (/usr/src/app/lib/ZwaveClient.js:278:26)
at ZWaveController.emit (node:events:390:22)
at ZWaveController.EventEmitter.emit (node:domain:470:12)
at ZWaveController.handleAddNodeRequest (/usr/src/app/node_modules/zwave-js/src/lib/controller/Controller.ts:1147:11)
at Driver.handleRequest (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:2169:22)
at Driver.handleUnsolicitedMessage (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1904:5)
```

I followed my steps from before:

1. Add my device (not added before)
2. Add my device again
  * To clarify: After step 1 is done:
  * I enable inclusion in the Z2M UI again
  * I tap my device to put it in include/exclude mode
3. Issue occurs.

**Comment by LordMike** [comment]
To be fair, I don't think this is a pressing issue - the device is added correctly the first time - the second time around is a state issue / unsupported behaviour. Devices are not meant to be included twice.
